North-Seattle-College/ad440-winter2022-thursday-repo
# Split large text into sentences and words

Closes #93 - Create a ML model to split the text in sentences

## Description

Created a class `Tokenizer` to be used by other scripts and/or API to split a large block of text into sentences, words, or to remove stopwords (such as "a", "the", "and", etc).

Returns the result in an array of strings

## Prerequisites

Install `nltk` by running:
 - `pip install nltk`

## Testing
- Make sure to `cd` into `/backend` in our repo
- run `python sentenceTokenizer.py` to run the script
- Some installation should happen in the background and download the necessary packages
![image](https://user-images.githubusercontent.com/62415298/156501367-3e7c669a-1751-47e7-b883-1897808c44e4.png)
- Your console/terminal will show a list of sentences from the predefined test text like below:
![image](https://user-images.githubusercontent.com/62415298/156501335-4d8070d5-6925-48d5-bf17-ec5ab000dabe.png)

**Additional testing**
- You may open the `sentenceTokenizer.py` file and change the string for `text` to do some additional testing.
- You may modify **line 42** in the file to `token.splitWords()` or `token.removeStopwords()` to see the different functions of the class

## Time estimate
Estimate: 4 hours
Actual: 10 hours

Hurdles: tried to deploy python class to AWS Lambda but ran into import issues, then API Gateway issues. Abandoned previous effort to do more research on how to deploy to Sagemaker instead.
